services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedCredentialsService.java

```diff
@@ -190,7 +208,7 @@ public MongoDbBasedCredentialsService(
         Objects.requireNonNull(deviceKey);
         Objects.requireNonNull(span);
 
-        return findCredentials(deviceKey)
+        return findCredentials(deviceKey, Optional.empty())
```

Comment:
why not *getCredentialsDto* and use the DTO's API to retrieve the credentials? As the name Data Transfer Object suggests, it should be used to transfer data to/from the persistence store, right? That way we also wouldn't require this method to have knowledge about the internal structure of the JSON document being stored in the DB ...

Reply:
@sophokles73 Sounds good. I have made those changes. Please take a look.